documentation/docs/examples/timeline.mdx

```diff
@@ -0,0 +1,153 @@
+
+:::note
+The test data used in the documentation site is for testing and to show the capabilities of the tool.
+The examples shown is therefore might not be a ideal model construction progression.
+:::
+
+<DemoWrapper name="Cognite3DViewerDemo" />
+
+## Timeline
+
+The timeline tool provides ability to apply styles by date. It works by creating a set of "keyframes"  for provided dates.
+Each keyframe has a set of associated styled node collections.
+
+### Creating Keyframes
+To create Keyframe, use `createKeyframe(date: Date)`
+- `date` - Date reference for the Keyframe. Should provide the date value by `Date.now()`/`new Date('2021-10-05')`.
+
+### Add Node collection & Styles to Keyframes
+Tool provides feature to add nodes & styles to a Keyframe or number of Keyframes or to all of the Keyframes.
+
+#### Set Styles for Node through Keyframes
+`assignStyledNodeCollection(keyframes: Keyframe | Keyframe[], nodeCollection: NodeCollectionBase, nodeAppearance: NodeAppearance)`
+- `keyframes` - List of Keyframe to apply Style.
+- `nodeCollection` - Set of nodes to be rendered for the Keyframe.
+- `nodeAppearance` - Styling of the nodes.
+
+### Play timelines
+We use `play(startDate: Date, endDate: Date, durationInMilliSeconds: number)`
+- `startDate` - Keyframe date to start the Playback of Timeline Keyframes.
+- `endDate` - Keyframe date to stop the Playback of Timeline Keyframes.
+- `durationInMilliSeconds` - Number of milli-seconds for the full plackback of Keyframes.
+
+```js runnable
+// import { NodeCollectionBase, DefaultNodeAppearance, PropertyFilterNodeCollection } from '@cognite/reveal';
+// import { TimelineTool, Keyframe } from '@cognite/reveal/tools';
+
+model.setDefaultNodeAppearance(DefaultNodeAppearance.Ghosted);
+
+const timeline = new TimelineTool(model);
+
+const nodes = new AssetNodeCollection(sdk, model);
+const keyframeAssetNode = timeline.createKeyframe(new Date('2021-10-05'));
+keyframeAssetNode.assignStyledNodeCollection(nodes, { renderGhosted: false, outlineColor: NodeOutlineColor.Cyan });
+
+nodes.executeFilter({ assetId: 4222791488928479 });
+const keyframeAssetID = timeline.createKeyframe(new Date('2021-10-06'));
+keyframeAssetID.assignStyledNodeCollection(nodes, { renderGhosted: false, outlineColor: NodeOutlineColor.Orange });
+
+const allPipes = new PropertyFilterNodeCollection(sdk, model);
+allPipes.executeFilter({'PDMS': {'Type': 'PIPE'}});
+const keyframeAllPipes = timeline.createKeyframe(new Date('2021-10-07'));
+keyframeAllPipes.assignStyledNodeCollection(allPipes, { renderGhosted: false, color: [40, 200, 20] });
+
+const stuMechNodes = new PropertyFilterNodeCollection(sdk, model);
+stuMechNodes.executeFilter({'PDMS': {':capDisc': 'MECH', 'Purpose': 'STU' }});
+const keyframeMechNodes = timeline.createKeyframe(new Date('2021-10-08'));
+keyframeMechNodes.assignStyledNodeCollection(stuMechNodes, { renderGhosted: false, color: [80, 100, 120] });
+
+timeline.play(new Date('2021-10-05'), new Date('2021-10-09'), 25000);
```

Comment:
It would be great if the example included using the callback for monitoring progress - e.g. just add a label to the 3D viewer that shows the current date or a progress bar, e.g. https://stackoverflow.com/a/7191095